### PR TITLE
add rule to check that else if blocks are followed by else blocks

### PR DESCRIPTION
**Pull request by mpokrass**
In response to augur bounty: https://augur.net/bounties/  for "An else clause must be included after else if "

### CODE REVIEW

File: rules/enforce-else-after-else-if.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+
+    create(context) {
+        function inspectElseIfStatement(node) {
+            // if the else if does not have an alternate, that means there is
+            // no else - report an error
+            if (!node['alternate']) {
+                context.report({
+                    node: node,
+                    message: 'else if statement not followed by else'
```
Comment (duaraghav8) [inline rules/enforce-else-after-else-if.js]: Please change message to `"else if" statement must be followed by an "else" statement`

File: test/enforce-else-after-else-if.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                "security/enforce-else-after-else-if": "error"
+        }
+};
+
+describe('[RULE] enforce-else-after-else-if: Rejections', function () {
+        it('should raise an error for an else if with no else after it', function(done) {
+                var code = [
+                        'contract Foo { function bar(n) { if (n > 10) { return n; } else if (n < 10) { return -n; } } }',
+                        'contract Foo { function bar(n) { if (n > 10) { return n; } else if (n < 10) { return -n; } else if (n == 0) { return 1; } } }',
+                        'contract Foo { function bar(n) { if (n > 10) { return n; } else if (n < 10) { return -n; } else { return 100; } } }',
```
Comment (duaraghav8) [inline test/enforce-else-after-else-if.js]: Please move the 3rd & 4th test (those who pass without any errors and therefore give `errors.length` = 0) to another `describe` section with description `[RULE] enforce-else-after-else-if: Acceptances`

**Comment by mpokrass** [comment]
@duaraghav8 fixed! let me know if there's anything else

**Comment by duaraghav8** [comment]
Thanks for your contribution @mpokrass :)

**Pull request merged**